src/main/java/deanery/Main.java

```diff
@@ -0,0 +1,61 @@
+package deanery;
+
+import deanery.exception.WrongHour;
+import deanery.exception.WrongNumber;
+import deanery.student.FullTimeStudent;
+import deanery.student.PartTimeStudent;
+import deanery.student.Student;
+
+import java.util.List;
+import java.util.Scanner;
+
+public class Main {
+    public static void main(String[] args) throws WrongNumber, WrongHour {
+        Scanner scanner = new Scanner(System.in);
+        List<Student> students = StudentData.createData();
+
+        for (int i = 0; i < students.size(); i++) {
+
+        try {
+            students.get(i).serve("D124", 19);
+        } catch (WrongNumber | WrongHour e) {
+            System.out.println("Exception " + e.getMessage());
+        }
+
+
+
+
+    }
+
+        System.out.println("Enter type of student: 'part time' 'full time' :");
+        String type = scanner.nextLine();
+        System.out.println("Enter student name:");
+        String name = scanner.next();
+        System.out.println("Enter student last name:");
+        String lastName = scanner.next();
+        System.out.println("Enter student ID: ");
+        Integer ID = scanner.nextInt();
+        System.out.println("Enter room:");
+        String room = scanner.next();
+        System.out.println("Enter hour");
+        Integer hour = scanner.nextInt();
+        if(type.equals("full time")){
+            FullTimeStudent s = new FullTimeStudent(name, lastName, ID);

```

Comment:
The idea here would be to first create the students and then show the situation where a student goes to the Deanery and gives their's ID.
And then the program would look for a student in a collection that holds them and call the serve method for that student.
You wouldn't have to provide the type, because the program would know whether this student is FullTime or PartTime due to inheritance features